oneclick.py

```diff
@@ -856,6 +867,9 @@ def update_metadata(self, catalog_item, identifier=None, metadata_replacement_po
             identifier, metadata, metadata_replacement_policy=metadata_replacement_policy
         )
 
+        if not isinstance(result, CoverageFailure):
+            self.handle_success(identifier)
+
```

Comment:
I was trying to keep the code that modifies the db together with the code that reports the db modified.  Under what circumstances would handle_success be called twice?